allocator.go

```diff
@@ -329,8 +340,8 @@ func (a *Allocator) Allocate() (ok bool, allocation Allocation) {
 	}
 
 	// set a hard time limit of 10s on our solver.
-	result := a.model.Solve(solver.WithTimeout(time.Second * 10))
-	if result.Infeasible() || result.Invalid() {
+	result := a.model.Solve(solver.WithTimeout(a.opts.searchTimeout))
+	if !(result.Feasible() || result.Optimal()) {
```

Comment:
The solver exhibits unexpected behaviour here. If the `.solve(...)` call times out, the solution is neither considered `valid, nor invalid, nor optimal, nor feasible`, hence it escaped the previous check and caused a `nil ref` error beneath, tighten the condition checking here to block this anomaly.